docs/3-write-ten-dapp/3-migration.md

```diff
@@ -0,0 +1,56 @@
+---
+sidebar_position: 3
+---
+
+# Migrate your dApp to TEN
+
+Migrating to TEN enables your dApp to leverage "Programmable Encryption". Below are steps to help you transition smoothly.
+
+### Key Migration Steps
+
+- Update your Hardhat deployment to support the `--network ten` option.
+- Add data protection logic to your view functions (if applicable).
+- Configure visibility rules for event logs and internal storage.
+- Add the TEN onboarding widget to your JavaScript UI.
+
+## 1. Configuring Hardhat
+
+First, set up a Hardhat project if you haven't already.
+
+### 1.1 Installing the TEN Hardhat Plugin
+
+To add TEN Network compatibility, install the `ten-hardhat-plugin`:
+
+```bash
+npm install ten-hardhat-plugin
+```
+
+_You can use `npm` or `yarn` to install plugins._
+
+### 1.2 Configuring `hardhat.config.js`
+
+Modify `hardhat.config.js` in your project’s root directory as follows:
+
+```javascript
+import { HardhatUserConfig } from "hardhat/config";
+import "@nomiclabs/hardhat-waffle";
+import "ten-hardhat-plugin";
+
+module.exports = {
+  solidity: "0.8.10",
+  networks: {
+    hardhat: {
+        // Configuration for the Hardhat Network
+    },
+    ten: {
+        url: "https://testnet.ten.xyz/v1/",
+        chainId: 443,
+        accounts: ["your-private-key"],
+    },
+  },
+};
+
+export default config;
+```
+
+Once configured, you can start writing or migrating your smart contracts.
```

Comment:
In the current docs, this page has lots of content.
We need to keep that content, but split it up in a page per feature. 
Will that go in this folder?